File: buildground-nexus/src/main/java/com/buildground/nexus/MainActivity.java
package com.buildground.nexus;

import android.app.Activity;
import android.graphics.Color;
import android.os.Bundle;
import android.view.Gravity;
import android.widget.LinearLayout;
import android.widget.TextView;

public final class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setGravity(Gravity.CENTER);
        root.setPadding(48, 48, 48, 48);
        root.setBackgroundColor(Color.parseColor("#1B1B1B"));

        TextView title = new TextView(this);
        title.setText("BUILDGROUND NEXUS");
        title.setTextColor(Color.parseColor("#FF7A00"));
        title.setTextSize(28f);
        title.setGravity(Gravity.CENTER);

        TextView status = new TextView(this);
        status.setText("Independent Core\nOffline by default\nBuildGround trust only");
        status.setTextColor(Color.parseColor("#F5F5F5"));
        status.setTextSize(16f);
        status.setGravity(Gravity.CENTER);
        status.setPadding(0, 28, 0, 0);

        root.addView(title);
        root.addView(status);
        setContentView(root);
    }
}

File: buildground-glasses-bridge/src/main/java/com/buildground/nexus/glasses/MainActivity.kt
package com.buildground.nexus.glasses

import android.app.Activity
import android.graphics.Color
import android.os.Bundle
import android.view.Gravity
import android.widget.LinearLayout
import android.widget.TextView

class MainActivity : Activity() {
    private lateinit var status: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            gravity = Gravity.CENTER_HORIZONTAL
            setPadding(32, 64, 32, 32)
            setBackgroundColor(Color.rgb(27, 27, 27))
        }

        root.addView(TextView(this).apply {
            text = "BUILDGROUND"
            textSize = 24f
            setTextColor(Color.rgb(255, 122, 0))
            gravity = Gravity.CENTER
        })

        root.addView(TextView(this).apply {
            text = "NEXUS GLASSES BRIDGE"
            textSize = 16f
            setTextColor(Color.rgb(245, 245, 245))
            gravity = Gravity.CENTER
            setPadding(0, 8, 0, 40)
        })

        status = TextView(this).apply {
            text = "Starting hardware bridge…"
            textSize = 14f
            setTextColor(Color.rgb(184, 184, 184))
            gravity = Gravity.CENTER
        }
        root.addView(status)
        setContentView(root)

        BuildGroundGlassesBridge.start { message ->
            runOnUiThread { status.text = message }
        }
    }
}
